csharp/test/Ice/maxDispatches/TestI.cs

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) ZeroC, Inc.
+
+#nullable enable
+
+namespace Ice.maxDispatches;
+
+internal sealed class TestIntfI : Test.TestIntfDisp_
+{
+    private readonly object _mutex = new();
+    private int _dispatchCount;
+    private int _maxDispatchCount;
+
+    public override async Task opAsync(Current current)
+    {
+        lock (_mutex)
+        {
+            if (++_dispatchCount > _maxDispatchCount)
+            {
+                _maxDispatchCount = _dispatchCount;
+            }
+        }
+        await Task.Delay(TimeSpan.FromMilliseconds(50));
```

Comment:
It would be best to not use a delay, for example use a monitor and signal it from an operation. This requires a separate OA.